File: src/main/java/io/github/mg138/tsbook/command/admin/Get.java
package io.github.mg138.tsbook.command.admin;

import io.github.mg138.tsbook.Book;

import org.bukkit.command.CommandSender;
import org.bukkit.configuration.file.YamlConfiguration;
import org.bukkit.entity.Player;

import static io.github.mg138.tsbook.command.admin.Give.getItem;

public class Get {
    public static boolean call(CommandSender sender) {
        sender.sendMessage(Book.Companion.getCfg().translate.translate("commands.feedback.get"));
        return true;
    }

    public static boolean call(CommandSender sender, String itemName) {
        if (!(sender instanceof Player)) {
            sender.sendMessage(Book.Companion.getCfg().translate.translate("errors.player_only"));
            return false;
        }
        Player player = (Player) sender;

        YamlConfiguration item = Book.Companion.getCfg().getItemConfig().getItemByID(itemName);
        if (item != null) {
            AdminCommands.setItem(itemName);
            sender.sendMessage(Book.Companion.getCfg().translate.translate("messages.get", (Player) sender));
            player.getInventory().addItem(getItem(Book.Companion.getCfg(), "item", item));
            return true;
        }

        sender.sendMessage(Book.Companion.getCfg().translate.translate("errors.item_not_found"));
        return false;
    }
}

File: src/main/kotlin/io/github/mg138/tsbook/command/admin/Get.kt
package io.github.mg138.tsbook.command.admin

import io.github.mg138.tsbook.item.attribute.DefaultIdentifier
import io.github.mg138.tsbook.setting.BookConfig
import io.github.mg138.tsbook.command.util.error.CommandError
import io.github.mg138.tsbook.command.util.CommandUtil
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

object Get {
    fun call(sender: CommandSender): Boolean {
        sender.sendMessage(BookConfig.language.commands.feedback.get)
        return true
    }

    fun call(sender: CommandSender, itemName: String): Boolean {
        if (sender !is Player) {
            CommandError.playerOnly(sender)
            return false
        }

        val item = CommandUtil.getItemByName(itemName, sender) ?: return false
        sender.sendMessage(CommandUtil.getGetMessage(sender.name, itemName))
        sender.inventory.addItem(CommandUtil.getItem(DefaultIdentifier.ITEM, item))
        return true
    }
}